wrappers/services.go

```diff
@@ -471,13 +471,48 @@ func tryFileUpdate(path string, desiredContent []byte) (old *osutil.MemoryFileSt
 	}
 }
 
+type ServiceQuotaMap map[*snap.AppInfo]*quota.Group
+
+// MakeServiceQuotaMap builds a map of services and their quota groups. The goal is
+// to make sure that if any quota group applies to a snap service, then we can easily
+// look it up in this based on it's snap.AppInfo. We take the snap instance name, the services
+// want to include in the map (should only be services of the snap name provided) and the quota
+// group of the snap.
+func MakeServiceQuotaMap(snapName string, svcs []*snap.AppInfo, grp *quota.Group) ServiceQuotaMap {
```

Comment:
once we solved the issue with AddSnapServices etc, this doesn't belong here but should be a unexported helper of servicestate

Reply:
We can't make it part of the servicestate package, this would result in cyclic dependency as it's used twice in services.go, not just in `AddSnapServices`, I belive it's for similar reason that `SnapServiceOptions` exists in the wrappers package.

Reply:
to be clear my problem with this here is not just a code org problem, it's conceptual, the idea is that servicestate sets up service options and wrappers follow them, here the two are mixed. Anyway I still think we can move this:

* as I said the call in AddSnapServices can be fixed by reworking backend/link.go which maybe you already did
*  the 2nd call is doing almost nothing, see my next comments

Reply:
I managed to get it seperated out and moved to helper.go in servicestate, however it *is exported* as it's used by test code and referenced by patch5.go